### ISSUE TITLE
No module named 'yaafelib'

### ISSUE THREAD

**Issue by NexiaMoe** [Ircam-WAM/TimeSide#215]
Hello, I'm trying to Run TimeSide on my wsl2 with docker desktop using wsl2, but there is error : 

- No Module Named 'yaafelib'
- FileNotFoundError: [Errno 2] No such file or directory: '/srv/app/node_modules'
- PermissionError: [Errno 13] Permission denied: '/srv/lib/timeside/tests/samples'

Do I missing something ?

I'm using Windows 10
WSL2 is Ubuntu 20.04 with Docker installed

``` 
Creating timeside_broker_1 ... done
Creating timeside_var_1    ... done
Creating timeside_db_1     ... done
Creating timeside_app_1    ... done
Creating timeside_worker_1 ... done
Attaching to timeside_var_1, timeside_broker_1, timeside_db_1, timeside_app_1, timeside_worker_1
broker_1  | 1:C 21 Jan 2021 07:41:44.744 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
broker_1  | 1:C 21 Jan 2021 07:41:44.744 # Redis version=5.0.10, bits=64, commit=00000000, modified=0, pid=1, just started
broker_1  | 1:C 21 Jan 2021 07:41:44.744 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
broker_1  | 1:M 21 Jan 2021 07:41:44.748 * Running mode=standalone, port=6379.
broker_1  | 1:M 21 Jan 2021 07:41:44.748 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
broker_1  | 1:M 21 Jan 2021 07:41:44.748 # Server initialized
broker_1  | 1:M 21 Jan 2021 07:41:44.748 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
db_1      |
db_1      | PostgreSQL Database directory appears to contain a database; Skipping initialization
db_1      |
db_1      | LOG:  database system was shut down at 2021-01-21 07:34:17 UTC
db_1      | LOG:  MultiXact member wraparound protections are now enabled
db_1      | LOG:  autovacuum launcher started
db_1      | LOG:  database system is ready to accept connections
broker_1  | 1:M 21 Jan 2021 07:41:44.749 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
timeside_var_1 exited with code 0
broker_1  | 1:M 21 Jan 2021 07:41:44.749 * Ready to accept connections
app_1     | Collecting youtube-dl
worker_1  | Collecting youtube-dl
app_1     |   Downloading youtube_dl-2021.1.16-py2.py3-none-any.whl (1.9 MB)
worker_1  |   Downloading youtube_dl-2021.1.16-py2.py3-none-any.whl (1.9 MB)
app_1     | Installing collected packages: youtube-dl
app_1     |   Attempting uninstall: youtube-dl
app_1     |     Found existing installation: youtube-dl 2020.3.1
worker_1  | Installing collected packages: youtube-dl
worker_1  |   Attempting uninstall: youtube-dl
worker_1  |     Found existing installation: youtube-dl 2020.3.1
app_1     |     Uninstalling youtube-dl-2020.3.1:
worker_1  |     Uninstalling youtube-dl-2020.3.1:
app_1     |       Successfully uninstalled youtube-dl-2020.3.1
worker_1  |       Successfully uninstalled youtube-dl-2020.3.1
app_1     | Successfully installed youtube-dl-2021.1.16
worker_1  | Successfully installed youtube-dl-2021.1.16
app_1     | WARNING: You are using pip version 20.2.3; however, version 20.3.3 is available.
app_1     | You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
worker_1  | WARNING: You are using pip version 20.2.3; however, version 20.3.3 is available.
worker_1  | You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
app_1     | [   INFO   ] MusicExtractorSVM: no classifier models were configured by default
worker_1  | [   INFO   ] MusicExtractorSVM: no classifier models were configured by default
app_1     | /srv/lib/timeside/timeside/core/tools/package.py:41: UserWarning: Failed loading timeside.plugins.analyzer.externals.yaafe (No module named 'yaafelib')
app_1     |   for importer, modname, ispkg
worker_1  | /srv/lib/timeside/timeside/core/tools/package.py:41: UserWarning: Failed loading timeside.plugins.analyzer.externals.yaafe (No module named 'yaafelib')
worker_1  |   for importer, modname, ispkg
timeside_worker_1 exited with code 1
app_1     | [   INFO   ] MusicExtractorSVM: no classifier models were configured by default
app_1     | /srv/lib/timeside/timeside/core/tools/package.py:41: UserWarning: Failed loading timeside.plugins.analyzer.externals.yaafe (No module named 'yaafelib')
app_1     |   for importer, modname, ispkg
app_1     | Operations to perform:
app_1     |   Apply all migrations: admin, auth, authtoken, contenttypes, django_celery_results, sessions, sites, timeside_server
app_1     | Running migrations:
app_1     |   No migrations to apply.
app_1     | [   INFO   ] MusicExtractorSVM: no classifier models were configured by default
app_1     | /srv/lib/timeside/timeside/core/tools/package.py:41: UserWarning: Failed loading timeside.plugins.analyzer.externals.yaafe (No module named 'yaafelib')
app_1     |   for importer, modname, ispkg
app_1     | [   INFO   ] MusicExtractorSVM: no classifier models were configured by default
app_1     | /srv/lib/timeside/timeside/core/tools/package.py:41: UserWarning: Failed loading timeside.plugins.analyzer.externals.yaafe (No module named 'yaafelib')
app_1     |   for importer, modname, ispkg
app_1     | Traceback (most recent call last):
app_1     |   File "/srv/app/manage.py", line 9, in <module>
app_1     |     execute_from_command_line(sys.argv)
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
app_1     |     utility.execute()
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/management/__init__.py", line 375, in execute
app_1     |     self.fetch_command(subcommand).run_from_argv(self.argv)
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/management/__init__.py", line 224, in fetch_command
app_1     |     klass = load_command_class(app_name, subcommand)
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/management/__init__.py", line 36, in load_command_class
app_1     |     module = import_module('%s.management.commands.%s' % (app_name, name))
app_1     |   File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
app_1     |     return _bootstrap._gcd_import(name[level:], package, level)
app_1     |   File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
app_1     |   File "<frozen importlib._bootstrap>", line 983, in _find_and_load
app_1     |   File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
app_1     |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
app_1     |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
app_1     |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
app_1     |   File "/srv/lib/timeside/timeside/server/management/commands/timeside-create-boilerplate.py", line 13, in <module>
app_1     |     from timeside.core.tools.test_samples import generateSamples
app_1     |   File "/srv/lib/timeside/timeside/core/tools/test_samples.py", line 448, in <module>
app_1     |     samples = generateSamples()
app_1     |   File "/srv/lib/timeside/timeside/core/tools/test_samples.py", line 245, in generateSamples
app_1     |     os.makedirs(samples_dir)
app_1     |   File "/usr/lib/python3.7/os.py", line 221, in makedirs
app_1     |     mkdir(name, mode)
app_1     | PermissionError: [Errno 13] Permission denied: '/srv/lib/timeside/tests/samples'
app_1     | [   INFO   ] MusicExtractorSVM: no classifier models were configured by default
app_1     | /srv/lib/timeside/timeside/core/tools/package.py:41: UserWarning: Failed loading timeside.plugins.analyzer.externals.yaafe (No module named 'yaafelib')
app_1     |   for importer, modname, ispkg
app_1     | [   INFO   ] MusicExtractorSVM: no classifier models were configured by default
app_1     | /srv/lib/timeside/timeside/core/tools/package.py:41: UserWarning: Failed loading timeside.plugins.analyzer.externals.yaafe (No module named 'yaafelib')
app_1     |   for importer, modname, ispkg
app_1     | Traceback (most recent call last):
app_1     |   File "/srv/app/manage.py", line 9, in <module>
app_1     |     execute_from_command_line(sys.argv)
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
app_1     |     utility.execute()
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/management/__init__.py", line 375, in execute
app_1     |     self.fetch_command(subcommand).run_from_argv(self.argv)
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/management/base.py", line 323, in run_from_argv
app_1     |     self.execute(*args, **cmd_options)
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/management/base.py", line 364, in execute
app_1     |     output = self.handle(*args, **options)
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 188, in handle
app_1     |     collected = self.collect()
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 105, in collect
app_1     |     for path, storage in finder.list(self.ignore_patterns):
app_1     |   File "/usr/local/lib/python3.7/dist-packages/npm/finders.py", line 93, in list
app_1     |     self.cached_list = list(self._make_list_generator(ignore_patterns))
app_1     |   File "/usr/local/lib/python3.7/dist-packages/npm/finders.py", line 100, in _make_list_generator
app_1     |     for path in get_files(storage, self.match_patterns, ignore_patterns):
app_1     |   File "/usr/local/lib/python3.7/dist-packages/npm/finders.py", line 49, in get_files
app_1     |     directories, files = storage.listdir(location)
app_1     |   File "/usr/local/lib/python3.7/dist-packages/django/core/files/storage.py", line 315, in listdir
app_1     |     for entry in os.scandir(path):
app_1     | FileNotFoundError: [Errno 2] No such file or directory: '/srv/app/node_modules'

```

**Comment by yomguy** [comment]
Hi @ajipw 

yaafelib has been dropped by default in the last version. We will maybe re-introduce it a later versions, but for now it only produces a warning.

The 2 next errors are related to file system permissions which would be dur to WSL in your case. What docker-compose command are you using?

**Comment by ajipw** [comment]
I just use `docker-compse up` , I try with ubuntu virtualbox it produce same error

**Comment by yomguy** [comment]
@ajipw it seems you use an old version, please try :
```
git checkout master
git pull
docker-compose pull
docker-compose up
```

**Comment by ajipw** [comment]
Hi, I Updated to the latest version, but I still can't access web server, it say :
`app_1     | FileNotFoundError: [Errno 2] No such file or directory: '/srv/app/node_modules'`

And if I access port 9080 my browser say ERR_EMPTY_RESPONSE

**Comment by yomguy** [comment]
Hi @NexiaMoe this has finally been fixed in 1.1
